docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+
+```shell
+dune --send-action hello hi '[ama]' hello@active
+```
+
+```txt
+executed transaction: 78ece3e63c3c6f11a210a9229f32b822c825fc72b268b1325b7dc1798797f460  104 bytes  431 us
+warning: transaction executed locally, but may not be confirmed by the network yet         ] 
+#         hello <= hello::hi                    {"nm":"ama"}
+>> Name : ama
+#         hello <= hello::gettime               ""
+>> Year: 2023, Month: 2, Day: 24, Hour: 0, Minute: 42, Second: 5
+```
+
+## Built-in Types
+
+Antelope supports several C++ data types for developing smart contracts. Developers can use these types to define data structures and write functions that interact with the EOS blockchain and smart contract system. Here are some of the most commonly used types:
+
+- `uint64_t`: This is an unsigned 64-bit integer type used for storing numeric values. It's commonly used for representing amounts of currency or other assets.
+- `name`: This is a type that represents an account name on the EOSIO network. Account names are 12-character strings that uniquely identify user accounts.
+- `asset`: This is a type that represents a quantity of a particular asset, such as EOS or another token. It includes both the amount and the symbol of the asset.
+- `string`: This is a type that represents a sequence of characters, such as a message or a username.
+- `time_point_sec`: This is a type that represents a point in time as the number of seconds since the Unix epoch (January 1, 1970).
+- `bool`: This is a type that represents a Boolean value, which can be either true or false.
+- `vector`: This is a type that represents a dynamic array of elements. It can be used to store lists of items such as account names or asset quantities.
+
+You can find a full list of all defined built-in types in the [abi_serializer.cpp file](https://github.com/AntelopeIO/leap/blob/a3e0756474a0899f94161b031a30bce4c496b292/libraries/chain/abi_serializer.cpp#L90-L128)
+
+The `hello` smart contract uses several built-in types:
+
+- `name` for the type of the `hi` action input parameter,
+- `uint64_t`, `uint32_t`, `uint16_t`, and `uint8_t` for the time sub-components.
+
+## Multi-index Tables
+
+In EOS smart contract development, a multi-index table is a database-like data structure that allows developers to store and manage data in a persistent and efficient manner. Multi-index tables are defined using the `eosio::multi_index` template class, and can store any number of rows, each of which contains a set of related data elements.
+
+Multi-index tables are often used in EOS smart contract development to store data that needs to be accessed frequently and efficiently, such as user account balances, transaction history, or other important information. Because multi-index tables are implemented as an on-chain database, they can be accessed and modified by other smart contracts, as well as by external applications through the EOSIO API.
```

Comment:
This is a long section, so it's important to cut down all fluff and make sure we're only saying what **needs** to be said. 

"Multi-index tables are often used in EOS smart contract development to store data that needs to be accessed frequently and efficiently, such as user account balances, transaction history, or other important information. Because multi-index tables are implemented as an on-chain database, they can be accessed and modified by other smart contracts, as well as by external applications through the EOSIO API." <-- This could be cut down quite a lot for example. 

Also, "they can be accessed and modified by other smart contracts, as well as by external applications through the EOSIO API." makes it sound like anyone can manipulate the data.

Reply:
reworked